src/EntityFramework/Migrations/Design/CSharpMigrationCodeGenerator.cs

```diff
@@ -1536,7 +1536,14 @@ protected virtual string TranslateColumnType(PrimitiveTypeKind primitiveTypeKind
         /// <returns> The quoted identifier. </returns>
         protected virtual string Quote(string identifier)
         {
-            return "\"" + identifier + "\"";
+            using (var writer = new StringWriter())
+            {
+                using (var provider = CodeDom.Compiler.CodeDomProvider.CreateProvider("CSharp"))
+                {
+                    provider.GenerateCodeFromExpression(new CodeDom.CodePrimitiveExpression(identifier), writer, null);
+                    return writer.ToString();
+                }
+            }
```

Comment:
Sorry for the delay - I have been using my version including CodeDOM for a while now and in general I think the CodeDOM approach works better when generating long strings etc. as it adds line breaks as required.
However, I did find an issue when generating migrations in a class library with a startup project which included the Microsoft.CodeDom.Providers.DotNetCompilerPlatform package as it resulted in not finding the correct version of the providers unless the package was also added to the class library. Although this did result in using the latest Roslyn based providers.
I am happy to switch to the 'simpler' Replace approach for the providers as suggested and will update my pull appropriately.